Sources/Extensions/UIKit/UIRefreshControlExtensions.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  UIRefreshControlExtensions.swift
+//  SwifterSwift
+//
+//  Created by ratul sharker on 7/24/18.
+//  Copyright © 2018 SwifterSwift
+//
+
+#if canImport(UIKit)
+import UIKit
+
+#if !os(watchOS)
+// MARK: - Methods
+public extension UIRefreshControl {
+
+    /// SwifterSwift: Programatically begin refresh control inside of UITableView.
+    ///
+    /// - Parameter tableView: UITableView instance, inside which the refresh control is contained.
+    public func beginRefreshProgramatically(in tableView: UITableView) {
```

Comment:
I feel like this is backwards and that the function should be a `UITableView` extension. I.e.
```Swift
extension UITableView {
    @available(iOS 10.0, *)
    public func refresh(animated: Bool = true) {
        guard let refreshControl = refreshControl else { return }

        // https://stackoverflow.com/questions/14718850/uirefreshcontrol-beginrefreshing-not-working-when-uitableviewcontroller-is-ins/14719658#14719658
        refreshControl.beginRefreshing()
        let offsetPoint = CGPoint(x: 0, y: -refreshControl.frame.height)
        setContentOffset(offsetPoint, animated: animated)
    }
}
```
You could potentially add this as an extension of `UIScrollView` instead, which is a superclass of `UITableView` that defines the property, but it doesn't feel right because the refresh of a scroll view may not necessarily be along the top like it is for table views.

You could, as an addition, add it to `UITableViewController` as well:
```Swift
extension UITableViewController {
    public func refresh(animated: Bool = true) {
        guard let refreshControl = refreshControl else { return }

        // https://stackoverflow.com/questions/14718850/uirefreshcontrol-beginrefreshing-not-working-when-uitableviewcontroller-is-ins/14719658#14719658
        refreshControl.beginRefreshing()
        let offsetPoint = CGPoint(x: 0, y: -refreshControl.frame.height)
        tableView.setContentOffset(offsetPoint, animated: animated)
    }
}
```

Notice that for `UITableViewController` you don't need `@available` because it's been available since iOS 6.

Reply:
I also had the similar feeling that the functionality should go with `UITableView` or `UIScrollView`. But in OOP way, which widget should get this functionality ?. This functionality belongs to `UIRefreshControl`. From developer (who are using SwifterSwift) perspective, it's more obvious to find this functionality in `UIRefreshControl`, which S/he attached to her/his `UITableView`/`UITableViewController`. 

Similarly this could also be a part of the `UITableViewController`, but in that way developer is expecting the functionality in the `UIRefreshControl`. If you check carefully i also kept the name similarity (original : `beginRefreshing`, extension: `beginRefreshProgramatically`). It happens to me every time typing the `tableView.reloadData()`, learning about the `tableView.reloadData { /*completion*/ }`, then using this functionality.

Another reason i didn't submit it as a `UITableView` extension is that, this would be only available for iOS 10+. To support iOS 8+ this extension should go with `UIRefreshControl`. Putting this functionality in `UITableViewController` won't completely solve the problem, to support `UITableView` scenario, the extension needed to be in `UITableView` or `UIRefreshControl`. If we choose `UITableView`, then the cases below iOS 10 require the same extension copied into `UIRefreshControl`. More over swifter swift is supporting iOS 8+.

Considering these cases, i ended up putting this extension into `UIRefreshControl` which could be used in all these scenarios.

Reply:
The problem is that you could be calling refresh for the wrong table view. E.g.
```Swift
let refreshControl1 = UIRefreshControl()
let refreshControl2 = UIRefreshControl()
let tableView = ...
tableView.refreshControl = refreshControl1
refreshControl2.beginRefreshProgramatically(in: tableView)
```
I know it's not likely to happen, you'd have to really mess it up, but it opens the door for a mistake like this.

I don't see why the functionality would belong to the refresh control instead of the table view. Since the function is only useful if the RC belongs to a TV, and since a RC can belong to anything while a TV can only have 1 RC, it seems to make a lot more sense for the function to be in the TV.

If you want to add pre-iOS 10 support it's not that hard.
```Swift
extension UITableView {
    public func refresh(animated: Bool = true) {
        let refreshControl: UIRefreshControl!
        if #available(iOS 10.0, *) {
            refreshControl = self.refreshControl
        } else {
            refreshControl = subviews.first(where: { $0 is UIRefreshControl }) as? UIRefreshControl
        }

        guard refreshControl != nil else { return }

        // https://stackoverflow.com/questions/14718850/uirefreshcontrol-beginrefreshing-not-working-when-uitableviewcontroller-is-ins/14719658#14719658
        refreshControl.beginRefreshing()
        let offsetPoint = CGPoint(x: 0, y: -refreshControl.frame.height)
        setContentOffset(offsetPoint, animated: animated)
    }
}
```

Reply:
😱 So much going on here